streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -568,6 +568,10 @@ void suspend(final boolean clean,
                 commit(false);
             } finally {
                 if (eosEnabled) {
+                    // In #commit, we don't do checkpoint file writing when eos is turned on. To avoid race condition when writing checkpoint file
```

Comment:
Nit: typos and comment suggestions:
```
// In #commit(), we don't write a checkpoint file when EOS is turned on.
// To avoid a race condition between writing and reading a checkpoint file,
// iff a task is migrated from one thread to another within the same instance,
// we need to always checkpoint offsets in #suspend() if EOS is enabled
// (instead of StateManager#closeSuspended()).
// Cf. https://issues.apache.org/jira/browse/KAFKA-7672
```